coalib/core/Core.py

```diff
@@ -327,9 +149,190 @@ def instantiate_and_track(prev_bear_type, next_bear_type):
+        """
+        # FIXME Allow to pass different executors nicely, for example to execute
+        # FIXME   coala with less cores, or to schedule jobs on distributed
+        # FIXME   systems (for example Mesos).
+
+        self._schedule_bears(self.bears_to_schedule)
+        try:
+            self.event_loop.run_forever()
+        finally:
+            self.event_loop.close()
+
+    def _schedule_bears(self, bears):
+        """
+        Schedules the tasks of bears.
+
+        :param bears:
+            A list of bear instances to be scheduled onto the process pool.
+        """
+        for bear in bears:
+            if self.dependency_tracker.get_dependencies(
+                    bear):  # pragma: no cover
+                logging.warning(
+                    'Dependencies for {!r} not yet resolved, holding back. '
+                    'This should not happen, the dependency tracking system '
+                    'should be smarter. Please report this to the developers.'
+                    .format(bear))
+            else:
+                tasks = {
+                    self.event_loop.run_in_executor(
+                        self.executor, bear.execute_task,
+                        bear_args, bear_kwargs)
+                    for bear_args, bear_kwargs in bear.generate_tasks()}
+
+                self.running_tasks[bear] = tasks
+
+                for task in tasks:
+                    task.add_done_callback(functools.partial(
+                        self._finish_task, bear))
+
+                logging.debug('Scheduled {!r} (tasks: {})'.format(bear,
+                                                                  len(tasks)))
+
+                if not tasks:
+                    # We need to recheck our runtime if something is left to
+                    # process, as when no tasks were offloaded the event-loop
+                    # could hang up otherwise.
+                    self._cleanup_bear(bear)
+
+    def _cleanup_bear(self, bear):
+        """
+        Cleans up state of an ongoing run for a bear.
+
+        - If the given bear has no running tasks left:
+          - Resolves its dependencies.
+          - Schedules dependant bears.
+          - Removes the bear from the ``running_tasks`` dict.
+        - Checks whether there are any remaining tasks, and quits the event loop
+          accordingly if none are left.
+
+        :param bear:
+            The bear to clean up state for.
+        """
+        if not self.running_tasks[bear]:
+            resolved_bears = self.dependency_tracker.resolve(bear)
+
+            if resolved_bears:
+                self._schedule_bears(resolved_bears)
+
+            del self.running_tasks[bear]
+
+        if not self.running_tasks:
+            # Check the DependencyTracker additionally for remaining
+            # dependencies.
+            resolved = self.dependency_tracker.are_dependencies_resolved
+            if not resolved:  # pragma: no cover
```

Comment:
Could you create a special pragma in setup.cfg for this.  We're quite close to removing the default 'no cover' pragma, replacing it with specific reasons.
I am pretty sure that we have a few other pragma's which exist because the code is apparently unreachable, and the message is very clearly "you've hit a bug; please report it".